Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -542,6 +542,26 @@ variable {R : Type u} [CommRing R] (E : EllipticCurve R)
 def j : R :=
   E.Δ'⁻¹ * E.c₄ ^ 3
 
+theorem j_eq_zero_iff' : E.j = 0 ↔ E.c₄ ^ 3 = 0 := by
+  simp only [EllipticCurve.j, Units.mul_right_eq_zero]
+
+alias ⟨_, j_eq_zero'⟩ := j_eq_zero_iff'
+
+theorem j_eq_zero_iff [IsReduced R] : E.j = 0 ↔ E.c₄ = 0 := by
+  rw [j_eq_zero_iff']
+  exact ⟨fun h ↦ IsNilpotent.eq_zero ⟨_, h⟩, fun h ↦ by simp [h]⟩
+
+theorem j_eq_zero : E.c₄ = 0 → E.j = 0 := fun h ↦ by
+  simp [j_eq_zero_iff', h]
+
+theorem j_ne_zero_iff' : E.j ≠ 0 ↔ E.c₄ ^ 3 ≠ 0 := not_congr E.j_eq_zero_iff'
```

Comment:
I'm also not convinced we need the contrapositive versions since you can just use `not_congr` whenever needed.

Reply:
Plain `not_congr` produces `¬ E.j = 0 ↔ ¬ E.c₄ ^ 3 = 0`, which cannot `rw` the `E.j ≠ 0`. So I prefer reserving the `ne` version.

Reply:
I'm still not convinced by this. Could you give me an example where you need the negative forms in e.g. #16864?

Reply:
> Could you give me an example where you need the negative forms

Sorry, but I don't have examples for now. Now #16864 does not depend on this PR anymore...

Reply:
Fine. I think I'll remove this.